pandas/core/dtypes/cast.py

```diff
@@ -1178,7 +1178,8 @@ def construct_1d_arraylike_from_scalar(value, length, dtype):
         subarr = DatetimeIndex([value] * length, dtype=dtype)
     elif is_categorical_dtype(dtype):
         from pandas import Categorical
-        subarr = Categorical([value] * length)
+        subarr = Categorical([value] * length, dtype.categories,
+                             ordered=dtype.ordered)
```

Comment:
I think this could be done a bit more directly with `Categorical([value] * length, dtype=dtype)`.

Reply:
Will do. I'll add a test for the Series constructor. @jschendel good point. I'll make that change as well.